sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/consumer_client.hpp

```diff
@@ -29,9 +32,17 @@ namespace Azure { namespace Messaging { namespace EventHubs {
      */
     Azure::Core::Http::Policies::RetryOptions RetryOptions{};
 
-    /**@brief  Message sender options.
-     */
-    Azure::Core::Amqp::_internal::MessageReceiverOptions ReceiverOptions{};
+    /** @brief Maximum message size for messages being sent. */
+    Azure::Nullable<std::uint64_t> MaxMessageSize;
+
+    /** @brief Name of the consumer client. */
+    std::string Name{};
+
+    /** @brief Settle mode used for completing transactions. */
+    Azure::Core::Amqp::_internal::ReceiverSettleMode SettleMode;
+
+    /** @brief The link name of this message consumer. */
+    Azure::Core::Amqp::Models::_internal::MessageTarget MessageTarget;
```

Comment:
This is a public options type exposing internal-only fields.

Reply:
QQ: How do you feel about `Azure::Core::_internal::ClientOptions`, which is the base class for public classes, exposing internal-only fields.